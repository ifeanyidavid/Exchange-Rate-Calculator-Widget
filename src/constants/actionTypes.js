export const FETCH_RATES_START = "FETCH_RATES_START";
export const FETCH_RATES_ERROR = "FETCH_RATES_ERROR";
export const FETCH_RATES_DONE = "FETCH_RATES_DONE";
export const UPDATE_SOURCE_CURRENCY = "UPDATE_SOURCE_CURRENCY";
export const UPDATE_DESTINATION_CURRENCY = "UPDATE_DESTINATION_CURRENCY";
export const EXCHANGE_CURRENCY = "EXCHANGE_CURRENCY";
