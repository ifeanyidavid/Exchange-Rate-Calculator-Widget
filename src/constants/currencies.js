export const currencies = {
  EUR: { id: 0, value: "EUR" },
  GBP: { id: 1, value: "GBP" },
  USD: { id: 2, value: "USD" }
};
