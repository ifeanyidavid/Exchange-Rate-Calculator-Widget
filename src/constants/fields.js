export const fieldNames = {
  sourceCurrency: "sourceCurrency",
  destinationCurrency: "destinationCurrency",
  sourceAmount: "sourceAmount",
  destinationAmount: "destinationAmount",
  balance: "balance",
  symbol: "symbol",
  description: "description",
  currency: "currency",
  id: "id",
  error: "error",
  loading: "loading",
  rates: "rates",
};
