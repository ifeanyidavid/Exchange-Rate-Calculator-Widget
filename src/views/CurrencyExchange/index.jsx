import React from "react";
import CurrencyExchange from "./CurrencyExchange";

const CurrencyExchangeWrapper = props => <CurrencyExchange {...props} />;
export default CurrencyExchangeWrapper;
