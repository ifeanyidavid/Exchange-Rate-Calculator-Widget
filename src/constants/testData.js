import { currencies } from "./currencies";
import { fieldNames } from "./fields";

export const testData = {
  rates: [
    {
      rates: { USD: 1.1075, GBP: 0.84195 },
      base: "EUR",
      date: "2019-12-09"
    },
    {
      rates: { EUR: 1.1877189857, USD: 1.3153987766 },
      base: "GBP",
      date: "2019-12-09"
    },
    {
      rates: { EUR: 0.9029345372, GBP: 0.7602257336 },
      base: "USD",
      date: "2019-12-09"
    }
  ],
  pocketsReducer: {
    transactionHistory: [
      {
        id: 0,
        description: `Exchanged to EUR`,
        time: new Date().toLocaleTimeString("en-GB"),
        sourceAmount: 23,
        destinationAmount: 42,
        sourceCurrency: "USD",
        destinationCurrency: "EUR"
      }
    ],
    pockets: {
      [currencies.EUR.value]: {
        [fieldNames.id]: currencies.EUR.id,
        [fieldNames.currency]: currencies.EUR.value,
        [fieldNames.balance]: 300,
        [fieldNames.symbol]: "€",
        [fieldNames.description]: "Euro"
      },
      [currencies.GBP.value]: {
        [fieldNames.id]: currencies.GBP.id,
        [fieldNames.currency]: currencies.GBP.value,
        [fieldNames.balance]: 300,
        [fieldNames.symbol]: "£",
        [fieldNames.description]: "British Pound"
      },
      [currencies.USD.value]: {
        [fieldNames.id]: currencies.USD.id,
        [fieldNames.currency]: currencies.USD.value,
        [fieldNames.balance]: 300,
        [fieldNames.symbol]: "$",
        [fieldNames.description]: "American Dollar"
      }
    }
  },
  sourceAmount: 23,
  destinationAmount: 54,
  sourceCurrency: "USD",
  destinationCurrency: "EUR",
  error: new Error("Network disconnected"),
  slideIndex: 1,
  pocketValues: [
    {
      [fieldNames.id]: currencies.EUR.id,
      [fieldNames.currency]: currencies.EUR.value,
      [fieldNames.balance]: 300,
      [fieldNames.symbol]: "€",
      [fieldNames.description]: "Euro"
    },
    {
      [fieldNames.id]: currencies.GBP.id,
      [fieldNames.currency]: currencies.GBP.value,
      [fieldNames.balance]: 300,
      [fieldNames.symbol]: "£",
      [fieldNames.description]: "British Pound"
    },
    {
      [fieldNames.id]: currencies.USD.id,
      [fieldNames.currency]: currencies.USD.value,
      [fieldNames.balance]: 300,
      [fieldNames.symbol]: "$",
      [fieldNames.description]: "American Dollar"
    }
  ]
};
