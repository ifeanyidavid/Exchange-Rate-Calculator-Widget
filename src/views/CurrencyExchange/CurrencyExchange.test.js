import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";
import toJson from "enzyme-to-json";
import CurrencyExchange from "./CurrencyExchange";
import { testData } from "../../constants/testData";

configure({ adapter: new Adapter() });

let [
  setExchangeOpenState,
  updateSourceCurrencySlide,
  updateDestinationCurrencySlide,
  doExchangeCurrency
] = new Array(4).fill(jest.fn());

function shallowSetup() {
  const pockets = testData.pocketValues;

  const transactionHistory = testData.pocketsReducer.transactionHistory;

  const mainPockets = testData.pocketsReducer.pockets;

  const rates = testData.rates;

  // sample props to pass to shallow render

  const props = {
    setExchangeOpenState: setExchangeOpenState,
    sourceCurrency: "EUR",
    destinationCurrency: "USD",
    pockets,
    rates,
    updateSourceCurrencySlide: updateSourceCurrencySlide,
    updateDestinationCurrencySlide: updateDestinationCurrencySlide,
    sourceConversionRates: 1,
    destinationConversionRates: 2,
    doExchangeCurrency: doExchangeCurrency,
    mainPockets,
    transactionHistory: transactionHistory
  };

  // wrapper instance around rendered output
  const currencyExchangeEnzymeWrapper = shallow(
    <CurrencyExchange {...props} />
  );

  return { props, currencyExchangeEnzymeWrapper };
}

describe("Currency exchange component", () => {
  it("Renders correctly", () => {
    const { currencyExchangeEnzymeWrapper } = shallowSetup();
    expect(toJson(currencyExchangeEnzymeWrapper)).toMatchSnapshot();
  });
});
