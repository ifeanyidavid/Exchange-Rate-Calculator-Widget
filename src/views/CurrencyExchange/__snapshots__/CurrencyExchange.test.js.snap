// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Currency exchange component Renders correctly 1`] = `
<Fragment>
  <ExchangeTopBar
    disableExchangeButton={[Function]}
    exhchangeCurrency={[Function]}
    setExchangeOpenState={[MockFunction]}
    sourceRate="1 EUR = 1.00 USD"
  />
  <ExchangeCurrencyCarousel
    currency="EUR"
    name="sourceAmount"
    onChange={[Function]}
    onSlide={[MockFunction]}
    pockets={
      Array [
        Object {
          "balance": 300,
          "currency": "EUR",
          "description": "Euro",
          "id": 0,
          "symbol": "€",
        },
        Object {
          "balance": 300,
          "currency": "GBP",
          "description": "British Pound",
          "id": 1,
          "symbol": "£",
        },
        Object {
          "balance": 300,
          "currency": "USD",
          "description": "American Dollar",
          "id": 2,
          "symbol": "$",
        },
      ]
    }
    ratesFetched={[Function]}
    recalculateOnSlide={[Function]}
    showExchangeRate={false}
    value=""
  />
  <hr />
  <ExchangeCurrencyCarousel
    currency="USD"
    destinationRate="1 USD = 2.00 EUR"
    name="destinationAmount"
    onChange={[Function]}
    onSlide={[MockFunction]}
    pockets={
      Array [
        Object {
          "balance": 300,
          "currency": "EUR",
          "description": "Euro",
          "id": 0,
          "symbol": "€",
        },
        Object {
          "balance": 300,
          "currency": "GBP",
          "description": "British Pound",
          "id": 1,
          "symbol": "£",
        },
        Object {
          "balance": 300,
          "currency": "USD",
          "description": "American Dollar",
          "id": 2,
          "symbol": "$",
        },
      ]
    }
    ratesFetched={[Function]}
    recalculateOnSlide={[Function]}
    showExchangeRate={true}
    value=""
  />
</Fragment>
`;
